src/Fixer/Strict/StrictTypesFixer.php

```diff
@@ -26,79 +27,51 @@
      */
     public function isCandidate(Tokens $tokens)
     {
-        return true;
+        return count($tokens) > 1 && $tokens[0]->isGivenKind(T_OPEN_TAG);
     }
 
     /**
      * {@inheritdoc}
      */
     public function fix(\SplFileInfo $file, Tokens $tokens)
     {
-        if (!$tokens->isMonolithicPhp()) {
-            return;
-        }
+        // check for if declaration already in file
+        $sequence = $this->getDeclareStrictTypeSequence();
+        $searchIndex = $tokens->getNextMeaningfulToken(0);
+        if (null === $searchIndex) {
+            $this->insertNewSequence($tokens); // not found, insert declare statement
 
-        $declareTokens = array(
-            new Token(array(T_OPEN_TAG, '<?php ')),
-            new Token(array(T_DECLARE, 'declare')),
-            new Token('('),
-            new Token(array(T_STRING, 'strict_types')),
-            new Token('='),
-            new Token(array(T_LNUMBER, '1')),
-            new Token(')'),
-            new Token(';'),
-            new Token(array(T_WHITESPACE, "\n\n")),
-        );
-
-        // check for valid construct, break on mismatch
-        for ($i = 0; ; ++$i) {
-            if (!$tokens[$i]->equals($declareTokens[$i], true)) {
-                break;
-            }
-            if ($i === count($declareTokens) - 1) {
-                return;
-            }
+            return;
         }
 
-        // empty file, just add the tokens
-        if (null === $tokens->getNextNonWhitespace(0)) {
-            $tokens->overrideRange(0, 0, $declareTokens);
+        $sequenceLocation = $tokens->findSequence($sequence, $searchIndex, null, false);
```

Comment:
@keradus we would a good speed gain here if we could use the `$end`. However the `$end` counts for each token of the sequence. This is very hard to figure out because we search for meaningful tokens only.
Shouldn't the `findSequence` - method only check the index of the first token of the sequence to be before the `$end` and not the other tokens?